packages/amplify-category-function/src/provider-utils/awscloudformation/utils/addLayerToFunctionUtils.ts

```diff
@@ -62,23 +63,34 @@ export const askLayerSelection = async (
   const askArnQuestion = layerSelections.includes(provideExistingARNsPrompt);
   layerSelections = layerSelections.filter(selection => selection !== provideExistingARNsPrompt);
 
+  let layerVersionArrPrompt: string[];
+
   for await (let layerName of layerSelections) {
-    const currentSelectionDefaults = filterProjectLayers(previousSelections).find(sel => sel.resourceName === layerName);
-    const currentVersion = currentSelectionDefaults ? currentSelectionDefaults.version.toString() : undefined;
     const layerVersions = await loadLayerDataFromCloud(context, layerName);
+    layerVersionArrPrompt = layerVersions.map(layerVersion => layerVersion.Version.toString());
+    if (!layerVersionArrPrompt.length) {
+      layerVersionArrPrompt.push('1'); // new latest version if no layers are pushed
```

Comment:
We cannot assume the version number, instead, skip the prompt and assume 'Always choose latest version'.

Versions cannot be pinned if they have not been deployed

Reply:
Also map will return an empty array so do proper check on length value not just the presence of the property.

Reply:
Added a check on Array to resolve this